src/org/webonise/multithreading/concurrentmodificationexception/Application.java

```diff
@@ -0,0 +1,39 @@
+package org.webonise.multithreading.concurrentmodificationexception;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class Application {
+    private List<Integer> list;
+
+    Application(){
+        this.list = new ArrayList<>();
+
+        for(int i=0; i<100; i++){
+            list.add(i);
+        }
+    }
+
+    public void start(){
+        iterateListAndLaunchThreadToModify();
+    }
+
+    private void iterateListAndLaunchThreadToModify(){
+
+        Iterator iterator = list.iterator();
+        Thread thread = new Thread(new WorkerThread(list));
+        thread.start();
+
+        try{
+            while (iterator.hasNext()){
+                System.out.println(iterator.next());
+                Thread.sleep(100);
+            }
+
+            thread.join();
```

Comment:
If main thread finishes before child threads, child threads will become orphans. This may cause JVM crashing in some older versions. That why it's safe to let child threads finish before main thread. Java The Complete Reference quotes this " for some older JVMs, if the main thread finishes before a child thread has completed, then the Java run-time system may “hang.” "